src/main/java/exceptions/ClosingSocketException.java

```diff
@@ -0,0 +1,8 @@
+package exceptions;
+
+public class ClosingSocketException extends RuntimeException {
+
+    public ClosingSocketException(String message) {
+        super("Socket could not close: " + message);
```

Comment:
The constructor of `RuntimeException` (_super_  in this case) can take a second argument called `cause`, which should take exception that triggered this one. You'll need to adjust the parameters :)